pkg/engine/task/task_apply.go

```diff
@@ -0,0 +1,137 @@
+package task
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	"github.com/kudobuilder/kudo/pkg/util/health"
+	apiextv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	apijson "k8s.io/apimachinery/pkg/util/json"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ApplyTask will apply a set of given resources to the cluster. See Run method for more details.
+type ApplyTask struct {
+	Name      string
+	Resources []string
+}
+
+// Run method for the ApplyTask. Given the task context, it renders the templates using context parameters
+// creates runtime objects and kustomizes them, and applies them using the controller client. Finally,
+// resources are checked for health.
+func (at ApplyTask) Run(ctx Context) (bool, error) {
+	// 1. - Render task templates -
+	rendered, err := render(at.Resources, ctx.Templates, ctx.Parameters, ctx.Meta)
+	if err != nil {
+		return false, fmt.Errorf("%wfailed to render task resources: %v", ErrFatalExecution, err)
```

Comment:
are these verbs right without the space?

Reply:
It doesn't matter where you put it, since `ErrFatalExecution` is an error with an empty string. For `fmt.Errorf` to work `%w` has to be in the error string once. If you think, it's weird, I can move them.